backend/executors/TasksDirectory.js

```diff
@@ -0,0 +1,874 @@
+
+            if (_.isNumber(prop.value)) {
+                updateLine += prop.key + '=' + prop.value + '\n';
+            } else {
+                updateLine += prop.key + '="' + prop.value + '"\n';
+            }
+        }
+    }
+
+    return updateLine;
+};
+
+var getCloudPropertiesUpdateLine = function (executionDetails, executionId) {
+    var updateLine = '';
+
+    if (executionDetails.SOFTLAYER) {
+        var username = executionDetails.SOFTLAYER.params.username;
+        var apiKey = executionDetails.SOFTLAYER.params.apiKey;
+        updateLine =
+            'user=' + username + '\n' +
+            'apiKey="' + apiKey + '"';
+    }
+    else if (executionDetails.HP) {
+        var key = executionDetails.HP.params.key;
+        var secretKey = executionDetails.HP.params.secretKey;
+        var project = executionDetails.HP.params.project;
+        updateLine =
+            'tenant="' + project + '"\n' +
+            'user="' + key + '"\n' +
+            'apiKey="' + secretKey + '"';
+        /*
+         'keyFile="' + newPemFile.getName() + '.pem"';
+         'keyPair="' + newPemFile.getName() + '"';
+         'securityGroup="' + cloudConf.securityGroup + '"';
+         */
+    } else if (executionDetails.EC2) {
+        var ec2user = executionDetails.EC2.params.apiKey;
+        var ec2apiKey = executionDetails.EC2.params.secretKey;
+        var ec2keyPair = executionDetails.EC2.params.keyPair.KeyName;
+
+        updateLine =
+            '\n\nuser="' + ec2user + '"\n' +
+            'apiKey="' + ec2apiKey + '"\n' +
+            'keyPair="' + ec2keyPair + '"\n' +
+            'keyFile="keyFile.pem"\n' +
+            'machineNamePrefix="cloudify-agent-widget-' + executionId + '"\n' +
+            'managementGroup="cloudify-manager-widget-' + executionId + '"\n';
+    }
+
+    updateLine += getRecipePropertiesUpdateLine(executionDetails);
+
+    return updateLine;
+};
+
+//---------------- COMMON TASKS ------------------------
+exports.common = {};
+exports.common.getWidget = function (executionModel, callback) {
+    logger.info('getting widget id ' + executionModel.getWidgetId());
+
+    managers.db.connect('widgets', function (db, collection, done) {
+        collection.findOne({_id: executionModel.getWidgetObjectId()}, function (err, result) {
+            if (err) {
+                logger.error('unable to find widget', err);
+                callback(err, executionModel);
+                done();
+                return;
+            }
+
+            if (!result) {
+                logger.error('result is null for widget find');
+                callback(new Error('could not find widget'), executionModel);
+                done();
+                return;
+            }
+
+            executionModel.setWidget(result);
+            callback(null, executionModel);
+            done();
+        });
+    });
+};
+
+exports.common.saveExecutionModel = function (executionModel, callback) {
```

Comment:
3 functions to update execution model.. sounds to me like they should be under "models.WidgetExecution" next to "User" and "RecipeType"